### PR TITLE
Some small improvements

### PR DESCRIPTION
**Pull request by KlossPeter**
Mostly clippy suggestions, some might improve perf (as removing unneeded clones), most should improve readability.

### CODE REVIEW

File: src/settings.rs [hunk 1]
```diff
@@ -37,7 +37,7 @@ impl SettingsFile {
                     Settings::parse(&content)?
                 } else {
                     needs_save = true;
-                    Default::default()
+                    Settings::default()
```
Comment (lzutao) [inline src/settings.rs]:
```suggestion
                    Self::default()
```

File: src/dist/download.rs [hunk 2]
```diff
@@ -59,8 +59,7 @@ impl<'a> DownloadCfg<'a> {
         let partial_file_path = target_file.with_file_name(
             target_file
                 .file_name()
-                .map(|s| s.to_str().unwrap_or("_"))
-                .unwrap_or("_")
+                .map_or("_", |s| s.to_str().unwrap_or("_"))
```
Comment (tesuji) [inline src/dist/download.rs]:
```suggestion
                .and_then(|s| s.to_str())
```

**Review by tesuji — COMMENTED** [review]
You need to run cargo fmt on finish.

File: src/dist/config.rs [hunk 3]
```diff
@@ -71,7 +71,7 @@ impl Config {
     }
 
     pub fn new() -> Self {
-        Default::default()
+        Config::default()
```
Comment (tesuji) [inline src/dist/config.rs]:
```suggestion
        Self::default()
```

File: src/dist/download.rs [hunk 4]
```diff
@@ -57,12 +57,7 @@ impl<'a> DownloadCfg<'a> {
         }
 
         let partial_file_path = target_file.with_file_name(
-            target_file
-                .file_name()
-                .map(|s| s.to_str().unwrap_or("_"))
-                .unwrap_or("_")
-                .to_owned()
-                + ".partial",
+            target_file.file_name().and_then(|s| s.to_str()).to_owned() + ".partial",
```
Comment (tesuji) [inline src/dist/download.rs]:
```suggestion
            target_file.file_name().and_then(|s| s.to_str()).unwrap("_").to_owned() + ".partial",
```

**Comment by kinnison** [comment]
These look pretty good, but I don't like a commit labelled `rustfmt` in the middle.  Could you please rebase, folding the formatting into the commits which need them?  Once that's done, I think this can be merged. :+1:

**Comment by kinnison** [comment]
Hi @KlossPeter 

Your branch needs rebasing onto current master as there are conflicts which need resolving.  In addition, could you please remove the formatting patch and instead ensure on each commit that `cargo fmt` has already been run?

You may find it easier to complete the rebase if you make many smaller commits.

Thanks,

Daniel.

**Comment by kinnison** [comment]
Hi @KlossPeter 

Have you run out of time to work on this, or decided to go and do something else?  It's no problem if you have, but if so, could you please close this PR?  If instead you want to fix the conflicts and clean up the commit history then I'd be pleased to look again at it.

**Comment by kinnison** [comment]
I'm having to assume this has been abandoned.

**Pull request closed without merging**